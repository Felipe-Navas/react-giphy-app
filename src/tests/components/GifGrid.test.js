import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Testing the <GifGrid /> component', () => {
  const category = 'bike';

  test('should render', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').text().trim()).toBe('Loading');
  });

  test('should render items when the useFetchGifs is called', () => {
    const gifs = [
      {
        id: '12345',
        url: 'https://localhost/index/img12345.jpg',
        title: 'Imagen 12345',
      },
      {
        id: '99999',
        url: 'https://localhost/index/img99999.jpg',
        title: 'Imagen 99999',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
