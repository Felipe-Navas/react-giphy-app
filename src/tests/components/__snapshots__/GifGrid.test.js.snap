// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <GifGrid /> component should render 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    bike
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing the <GifGrid /> component should render items when the useFetchGifs is called 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    bike
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="12345"
      key="12345"
      title="Imagen 12345"
      url="https://localhost/index/img12345.jpg"
    />
    <GifGridItem
      id="99999"
      key="99999"
      title="Imagen 99999"
      url="https://localhost/index/img99999.jpg"
    />
  </div>
</Fragment>
`;
